pkg/datapath/loader/netlink.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loader
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	libbpfFixupMsg = "struct bpf_elf_map fixup performed due to size mismatch!"
+
+	// ExecTimeout is the execution timeout to use in join_ep.sh executions
+	ExecTimeout = 300 * time.Second
+)
+
+func replaceQdisc(ifName string) error {
+	link, err := netlink.LinkByName(ifName)
+	if err != nil {
+		return err
+	}
+	attrs := netlink.QdiscAttrs{
+		LinkIndex: link.Attrs().Index,
+		Handle:    netlink.MakeHandle(0xffff, 0),
+		Parent:    netlink.HANDLE_CLSACT,
+	}
+
+	qdisc := &netlink.GenericQdisc{
+		QdiscAttrs: attrs,
+		QdiscType:  "clsact",
+	}
+
+	// Ignore failure on replacing clsact, means we replaced it.
```

Comment:
This sentence doesn't make sense to me. Even if QdiscReplace returns an error it meas we have replace it? :thinking:

Reply:
Let me double-check this. We don't check the error today, probably because it can return an error if there's already a qdisc there (for example, because the qdisc we want to install already exists).

Furthermore, this should succeed the first time because *we* create the link. So it should only end up in a bad state if someone's really messing around with the devices.

Reply:
OK, I traced this down to the netlink library inserting an extra `TCA_OPTIONS` option to the command with empty contents, which is unnecessary and messes with the replacement of the qdisc. :/